07-routing-gossip.md

```diff
@@ -510,16 +524,197 @@ An endpoint node:
+queries for gossip synchronization.  These explicitly request what
+gossip should be received.
+
+There are several messages which contain a long array of
+`short_channel_id`s (called `encoded_short_ids`) so we utilize a
+simple compression scheme: the first byte indicates the encoding, the
+rest contains the data.
+
+Encoding types:
+* `0`: uncompressed array of `short_channel_id` types, in ascending order.
+
+### The `query_short_channel_ids`/`reply_short_channel_ids_done` Messages
+
+1. type: 261 (`query_short_channel_ids`) (`gossip_queries`)
+2. data:
+    * [`32`:`chain_hash`]
+    * [`2`:`len`]
+    * [`len`:`encoded_short_ids`]
+
+1. type: 262 (`reply_short_channel_ids_end`) (`gossip_queries`)
+2. data:
+    * [`32`:`chain_hash`]
+    * [`1`:`complete`]
+
+This is general mechanism which lets a node query for
+`channel_announcement` and `channel_update`s for specific shortids;
+usually either because it sees a `channel_update` for which it has no
+`channel_announcement` or because it has obtained them from
+`reply_channel_range`.
+
+#### Requirements
+
+The sender:
+  - MUST NOT send this if it has sent a previous `query_short_channel_ids` to this peer and not received `reply_short_channel_ids_end`.
+  - MUST set `chain_hash` to the 32-byte hash that uniquely identifies the chain
+  that the `short_channel_id`s refer to.
+  - MUST set the first byte of `encoded_short_ids` to zero
+  - MUST append a whole number of `short_channel_id`s to `encoded_short_ids`
+  - MAY send this if it receives a `channel_update` for a
+   `short_channel_id` for which has no `channel_announcement`.
+  - SHOULD NOT send this if the channel referred to is not an unspent output.
+
+The receiver:
+  - if the first byte of `encoded_short_ids` is not zero:
+    - MAY fail the connection
+  - if `encoded_short_ids` does not decode into a whole number of `short_channel_id`:
+    - MAY fail the connection.
+  - if it has not sent `reply_short_channel_ids_end` to a previously received `query_short_channel_ids` from this sender:
+    - MAY fail the connection.
+  - MUST respond to each known `short_channel_id` with a `channel_announcement`
+    and the latest `channel_update`s for each end
+	- SHOULD NOT wait for the next outgoing announcement flush to send these.
+  - MUST follow with any `node_announcement`s for each `channel_announcement`
+	- SHOULD avoid sending duplicate `node_announcements` in response to a single `query_short_channel_ids`.
+  - MUST follow these responses with `reply_short_channel_ids_end`.
+  - if does not maintain up-to-date channel information for `chain_hash`:
+	- MUST set `complete` to 0.
+  - otherwise:
+	- SHOULD set `complete` to 1.
+
+#### Rationale
+
+Future nodes may not have complete information; they certainly won't have
+complete information on unknown `chain_hash` chains.  While this `complete`
+field cannot be trusted, if 0 ir does indicate that the sender should search
```

Comment:
typo: s/ir/it/. I would actually just remove the "if" and "it": "While this `complete` field cannot be trusted, 0 does indicate...".

Reply:
Yep, will fix.